src/Raven.Server/Rachis/Leader.cs

```diff
@@ -801,6 +801,11 @@ public bool TryModifyTopology(string nodeTag, string nodeUrl, TopologyModificati
                 {
                     Interlocked.Exchange(ref _topologyModification, null);
                 });
+
+                if (modification == TopologyModification.Remove)
+                {
+                    EnsureNodeRemovalOnDeletion(nodeTag, context);
```

Comment:
Move this into the state machine, we want to be able at some point to use this for other stuff rather than just the cluster.